core/src/utils/input-shims/hacks/scroll-assist.ts

```diff
@@ -181,6 +181,24 @@ const setManualFocus = (el: HTMLElement) => {
     return;
   }
 
+  /**
+   * Optimization for scenarios where the currently focused element is a sibling
+   * of the target element. In such cases, we avoid setting `SKIP_SCROLL_ASSIST`.
+   *
+   * This is crucial for accessibility: input elements can now contain focusable
+   * siblings (e.g., clear buttons, slotted elements). If we didn't skip setting
+   * the attribute here, screen readers would be unable to navigate to and interact
+   * with these sibling elements.
+   *
+   * Without this check, we would need to call `ev.stopPropagation()` on the
+   * 'focusin' event of each focusable sibling to prevent the scroll assist
+   * listener from incorrectly moving focus back to the input. This approach
+   * is less maintainable and more error-prone.
```

Comment:
```suggestion
   * listener from incorrectly moving focus back to the input. That approach
   * would be maintainable and more error-prone.
```

Technically I think the way it is now is fine, just a little confusing. I tried to clear up confusion, but feel free to disregard this suggestion